templates/ray-data-llm/README.md

```diff
@@ -0,0 +1,162 @@
+
+
+
+```python
+# Prompts the user for Hugging Face token if required by the model.
+from util.utils import prompt_for_hugging_face_token
+HF_TOKEN = prompt_for_hugging_face_token("meta-llama/Meta-Llama-3.1-8B-Instruct")
+```
+
+In this example, we will be using the `meta-llama/Meta-Llama-3.1-8B-Instruct` model.
+We will also need to define a configuration associated with the model we want to use to configure the compute resources, engine arguments and other inference engine specific parameters. For more details on the the model configs, see the [API doc](https://docs.anyscale.com/llms/serving/guides/bring_any_model/) on bringing your own models.
+
+
+```python
+from ray.data.llm import vLLMEngineProcessorConfig
+from util.utils import is_on_gcp_cloud
+
+# There's no a10g on GCP.
+accelerator_type = "L4" if is_on_gcp_cloud() else "A10G"
+
+processor_config = vLLMEngineProcessorConfig(
+    model_source="meta-llama/Meta-Llama-3.1-8B-Instruct",
+    engine_kwargs=dict(
+        tensor_parallel_size=1,
+        pipeline_parallel_size=1,
+        max_model_len=16384,
+        enable_chunked_prefill=True,
+        max_num_batched_tokens=2048,
+    ),
+    # Override Ray's runtime env to include the Hugging Face token. Ray is being used under the hood to orchestrate the inference pipeline.
+    runtime_env=dict(
+        env_vars=dict(
+            HF_TOKEN=HF_TOKEN,
+        ),
+    ),
+    batch_size=16,
+    accelerator_type=accelerator_type,
+)
+
+```
+
+## Step 3: Define the preprocess and postprocess lambda
+
+
+We will need to define the preprocess lambda to convert input dataset to format that `vLLMEngineProcessor` can consume, and also postprocessor
+lambda that filter out the uninterested fields from vLLM engine.
+
+
+```python
+preprocess = lambda row: dict(
+    messages=[
+        {
+            "role": "system",
+            "content": "You are a commentator. Your task is to "
+            "summarize highlights from article.",
+        },
+        {
+            "role": "user",
+            "content": f"# Article:\n{row['article']}\n\n"
+            "#Instructions:\nIn clear and concise language, "
+            "summarize the highlights presented in the article.",
+        },
+    ],
+    sampling_params=dict(
+        temperature=0.3,
+        max_tokens=150,
+        detokenize=False,
+    ),
+)
+postprocess = lambda row: {
+    "resp": row["generated_text"],
+}
+```
+
+## Step 4: Build and run the processor
+
+
+With the workload and configs defined, we can now build then run the processor
+
+
+```python
+from ray.data.llm import build_llm_processor
+
+processor = build_llm_processor(
+    processor_config,
+    preprocess=preprocess,
+    postprocess=postprocess,
+)
+
+ds = processor(ds)
+ds = ds.materialize()
```

Comment:
Add a comment saying that this could be any data writing APIs (e.g. `write_parquet`) and point to the API doc.